CONTRIBUTING.md

```diff
@@ -23,6 +23,102 @@ we'll be able to accept your pull requests.
 [individual-cla]: https://developers.google.com/open-source/cla/individual
 [corporate-cla]: https://developers.google.com/open-source/cla/corporate
 
+Code Layout
+===
+
+The source code is divided into subdirectories based on its purpose:
+
+- `src`: the main API code, shipped as NuGet packages
+- `test`: tests of the code in `src`, both unit tests and
+  integration tests
+- `snippets`: small pieces of code used to add examples to the API
+  reference and demonstrate the functionality available. These are
+  written in the form of integration tests, but are not expected
+  to fully test the functionality of the API. Instead, the assertions
+  are really present to ensure that the snippets themselves are
+  correct.
+- `tools`: tools used within this project. These are not shipped
+  externally, and not expected to be used by developers who only
+  need to access the Google APIs covered in this repository.
+
+Running the tests
+---
+
+As noted above, both the `snippets` directory and the `test`
+directory contain tests, of different flavours. All tests use
+[xUnit.Net](http://xunit.github.io/). Unit tests do not require
+any access to Google Cloud Platform, but the integration tests
+and snippet tests require a Google Cloud Platform project with
+billing enabled.
+
+**Please note that running the tests exercises the real API
+against real Google Cloud Platform projects.** The tests are
+designed not to require too much in the way of resources, but you
+should be aware that there *is* a cost to exercising APIs. In future
+we may run tests against emulators where available. We **strongly**
+advise you to use a dedicated testing project (rather than a
+project relied on by production systems) to avoid any possible data loss.
+
+To run integration tests with your project, you need to perform some
+set-up steps:
+
+- Ensure that your project has the corresponding API enabled in
+  the [API Manager](https://console.developers.google.com/apis/library)
+- Download and install the [https://cloud.google.com/sdk/](Cloud SDK)
+- Run `gcloud auth login` to set up default application credentials
+- Set the `TEST_PROJECT` environment variable to the name of your project.
+  To run the tests from a command line, you can just set them directly
+  and run the tests. To run the tests from Visual Studio, you should set
+  the environment variable permanently, and restart Visual Studio if
+  it's already running.
+- Build all the code, either from Visual Studio, or from the command line
+  with `dnu`/`dnx`.
```

Comment:
would it be possible to see the specific sequence of commands here that build and test the unit tests and that build and test the snippet tests?